### ISSUE TITLE
Lists in Markdown don't render/bullets disappear

### ISSUE THREAD

**Issue by rghedin** [Automattic/jetpack#39605]
### Impacted plugin

Jetpack

### Quick summary

When writing a comment, a list made of &ast; or - not only doesn't render properly as a HTML list, but also hides those characters. (At least using default comment form.)

### Steps to reproduce

1. Activate Markdown module on Jetpack, and only it.
2. Browse a blog post with comments open.
3. Type a list using &ast; or -.

### A clear and concise description of what you expected to happen.

* Item 1
* Item 2
* Item 3

### What actually happened

Item 1
Item 2
Item 3

### Impact

Most (> 50%)

### Available workarounds?

No but the platform is still usable

**Comment by phcp** [comment]
📌 REPRODUCTION RESULTS

Replicated only with comments on Atomic sites without using a block-based comment box. I couldn't spot any quick fix other than extending the existing [markdown parser](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/markdown/easy-markdown.php#L185) used for comments.

📌 FINDINGS/SCREENSHOTS/VIDEO

<img src="https://github.com/user-attachments/assets/5b7b3b07-ee3c-4541-af53-a7bd114b6e12" width="600">
<img src="https://github.com/user-attachments/assets/bccf87c3-90cb-43a3-b04a-63af59244b2e" width="400">

📌 ACTIONS

- Triaged;
- Priority reduced to Normal since the comments are still usable, and as a workaround, it's possible to edit as an HTML when the list style is needed;

**Comment by rghedin** [comment]
Have you tried edit the comment to use HTML, @phcp? In my site, this workaround doesn't work either. (Although I'm not sure if it's due to some CSS mistake.)

**Comment by phcp** [comment]
> Have you tried edit the comment to use HTML, @phcp? In my site, this workaround doesn't work either. (Although I'm not sure if it's due to some CSS mistake.)

Yes, it worked fine for me. I had to submit the comment, click Edit, and format it as an HTML list.

<img src="https://github.com/user-attachments/assets/dc2e64b4-6a99-4473-ae89-30d9ae1a3a65" width="400" />
<img src="https://github.com/user-attachments/assets/e2fa05ce-67f6-4737-8eaa-16f548c23d73" width="400" />

**Comment by jeherve** [comment]
Could you go to Jetpack > Settings, search for "markdown", and ensure Markdown is enabled for comments? Once you turn that on, you should see Markdown being processed in your future comments.

Let me know how that goes!

**Comment by rghedin** [comment]
It is, @jeherve. Markdown works fine in comments, except for lists.

![Image](https://github.com/user-attachments/assets/93857d76-1735-4003-9bcb-c9823cce8a8c)

**Comment by jeherve** [comment]
Thanks for giving it a try! We'll need to look at how data is sanitized by WordPress when submitting a comment. I suspect some HTML, like list items, are getting stripped by WordPress, but we'll see what we can do to fix that!

**Comment by rghedin** [comment]
Bug still present.
